hardware/swabian_instruments/pulse_streamer.py

```diff
@@ -25,107 +20,167 @@
 top-level directory of this distribution and at <https://github.com/Ulm-IQO/qudi/>
 """
 
+
 from core.module import Base
 from core.configoption import ConfigOption
+from core.statusvariable import  StatusVar
 from core.util.modules import get_home_dir
 from interface.pulser_interface import PulserInterface, PulserConstraints
 from collections import OrderedDict
+import numpy as np
 
-import grpc
-import os
-import hardware.swabian_instruments.pulse_streamer_pb2 as pulse_streamer_pb2
-import dill
+import pulsestreamer as ps
 
 
 class PulseStreamer(Base, PulserInterface):
-    """ Methods to control PulseStreamer.
-
-    Example config for copy-paste:
-
-    pulse_streamer:
-        module.Class: 'swabian_instruments.pulse_streamer.PulseStreamer'
-        pulsestreamer_ip: '192.168.1.100'
-        laser_channel: 0
-        uw_x_channel: 2
-
+    """ Interface class to define the abstract controls and
+    communication with all pulsing devices.
     """
 
     _pulsestreamer_ip = ConfigOption('pulsestreamer_ip', '192.168.1.100', missing='warn')
-    _laser_channel = ConfigOption('laser_channel', 0, missing='warn')
-    _uw_x_channel = ConfigOption('uw_x_channel', 2, missing='warn')
+    _laser_channel = ConfigOption('laser_channel', 1, missing='warn')
+    _uw_x_channel = ConfigOption('uw_x_channel', 3, missing='warn')
+    _use_external_clock = ConfigOption('use_external_clock',False,missing='warn')
```

Comment:
Not using an optional external clock input should not trigger a warning. (info level at most)

Reply:
fixed